src/coreclr/jit/compiler.cpp

```diff
@@ -3236,6 +3240,12 @@ void Compiler::compInitOptions(JitFlags* jitFlags)
     }
 
 #ifdef DEBUG
+    // JitFakeProcedureSplitting overrides JitNoProcedureSplitting with a fake splitting implementation
+    if (JitConfig.JitFakeProcedureSplitting())
+    {
+        opts.compProcedureSplitting = true;
+    }
+
```

Comment:
I don't understand why this is necessary. If someone sets `COMPlus_JitFakeProcedureSplitting=1` and `COMPlus_JitNoProcedureSplitting=Main`, it seems like that should work: do fake splitting on all functions except `Main`. 

It seems like the real issue is:
```
 opts.compProcedureSplitting = !opts.compDbgCode;
```
maybe should be:
```
if (opts.compDbgCode && !JitConfig.JitFakeProcedureSplitting())
{
 opts.compProcedureSplitting = false;
}
```

Reply:
I see, looks like I misunderstood what precedence JitFakeProcedureSplitting should (not) take over other configurations. I've changed the logic so JitFakeProcedureSplitting can be partially overriden by other configs, but can still override opts.compDbgCode if enabled.